src/Testers/WarehouseTest.java

```diff
@@ -47,4 +51,15 @@ public void checkIfEmptyMapThrowMessageTest() {
 
         assertEquals(actualMessage, expectedMessage);
     }
+    private Product test1 = new Product(KindOfProduct.SMARTPHONE, "Apple", "iPhone 12", "ES description", 2, 12, 1000, 1250);
+    @Test
+    public void removeProduct_WhenStockIsNotEmpty() {
+        Warehouse warehouse = new Warehouse();
+        fillUpWarehouse(warehouse);
+        int initialQuantity = 10 ;
+        int quantityToRemove = 9 ;
+       warehouse.addProduct(test1 , initialQuantity);
+       warehouse.removeProduct(test1 , quantityToRemove);
+       assertEquals(initialQuantity - quantityToRemove , warehouse.getStock().get(test1).intValue());
```

Comment:
initialQuantity - quantityToRemove-> 1

Non vedo perché mettere la sottrazione delle variabili in questo caso, visto che sono state inizializzate nel metodo, non possono che risultare 1. Sarebbe stato diverso se i valori fossero stati presi nella definizione del metodo